src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/PlatformEqtTrace.cs

```diff
@@ -147,6 +175,11 @@ public bool InitializeVerboseTrace(string customLogFile)
         /// <inheritdoc/>
         public bool ShouldTrace(PlatformTraceLevel traceLevel)
         {
+            if (DoNotInitialize)
+            {
+                return false;
+            }
+
             switch (traceLevel)
             {
                 case PlatformTraceLevel.Off:
```

Comment:
<GitHub does not let me add comments to the line I want :-/ >
If I understand the reason for this change - Calling `ShouldTrace` even if `PlatformTraceLevel.Off` is the target, we end up initializing the Source instance since it is static. Can we just pull that out to a separate method instead please? You would not need a DoNotInitialize then.
That apart, should we be caching Source.Switch.ShouldTrace value and reuse?